sdk/runtime/runner.go

```diff
@@ -5,13 +5,43 @@ import (
 	"io"
 	"os"
 	"strings"
+	"time"
 )
 
 // Invoke runs the passed test-case and reports the result.
 func Invoke(tc func(*RunEnv) error) {
 	runenv := CurrentRunEnv()
+	start := time.Now()
+	durationGuage := NewGauge(runenv, "plan_duration", "Run time (Seconds)")
+
+	// The prometheus pushgateway has a customized scrape interval, which is used to hint to the
+	// prometheus operator at which interval the it should be scraped. This is currently set to 5s.
+	// To provide an updated metric in every scrape, jobs will push to the pushgateway at the same
+	// interval. When this "push_interval" is changed, you may want to change the scrape interval
+	// on the pushgateway
+	push_interval := 5
```

Comment:
I *am* worried about the scalability when we have a high number of nodes.

I think the nodes will be started fairly unevenly, so the pushes will come in probably evenly-distributed, but the load of the pushgateway would grow with the size of the cluster. A test with 1,000 instances will cause the pushgateway to take on 250 PUT requests per second. Although I think we can deal with this when it comes up.